### PR TITLE
azure_rm_virtualmachine - Enhance support for azure vms created from snapshots

### PR DESCRIPTION
**Pull request by yfried**
##### SUMMARY
Module is crashing with KeyError for some attributes that don't exit for VMs that were created from disk snapshots

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
azure_rm_virtualmachine

##### ANSIBLE VERSION

2.7
##### ADDITIONAL INFORMATION
Follow up on #45269

### CODE REVIEW

**Comment by jhawkesworth** [comment]
closing and opening this PR to re-trigger tests (not sure if the shippable failures are transient).

**Comment by Fred-sun** [comment]
@yfried Thanks for your contribution, Could you help resolve the conflicts files? when you finish changed. I will try crashing issue again . Thanks!

**Comment by yfried** [comment]
> @yfried Thanks for your contribution, Could you help resolve the conflicts files? when you finish changed. I will try crashing issue again . Thanks!

@Fred-sun Done tnx.

**Comment by Fred-sun** [comment]
@zikalino Could you help take a look this PR? Thanks a lot!

**Comment by yungezz** [comment]
The change looks good. Could you pls update existing tests to cover validation on the osProfile and imageReference?

**Comment by yfried** [comment]
> The change looks good. Could you pls update existing tests to cover validation on the osProfile and imageReference?

@yungezz How do I do that? where are the relevant tests?

**Comment by Fred-sun** [comment]
@yungezz Please help take a look? Thanks!

**Comment by Fred-sun** [comment]
@yungezz @zikalino  The contributor need help from your. Please help. Thanks!

**Comment by Fred-sun** [comment]
@yungezz @zikalino  Could you help this ?

**Comment by Fred-sun** [comment]
@yungezz @zikalino

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py [hunk 1]
```diff
@@ -1230,17 +1233,17 @@ def exec_module(self, **kwargs):
                         vm_resource.tags = vm_dict['tags']
 
                     # Add custom_data, if provided
-                    if vm_dict['properties']['osProfile'].get('customData'):
+                    if vm_dict['properties'].get('osProfile', {}).get('customData'):
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py]: pls, undo this change. as far as I remember, for VMs created from existing os disks it may cause crash

**Comment by Fred-sun** [comment]
@yfried Thanks for your contribution. Could you can resolve the conflicting file and move to review ? Thanks!

**Comment by Fred-sun** [comment]
@yfried Are you still here? Could you can resolve the conflicting file and move to review ? Thanks!

**Comment by yfried** [comment]
@Fred-sun Looking at the new module, looks like my changes are no longer needed. I think this PR can be closed and the issue resolved

**Comment by Fred-sun** [comment]
@yfried Thank you for your update. Can you close this PR?

**Comment by Fred-sun** [comment]
closed_me, It has been fixed by latest module! Thanks!
